src/OpenTelemetry.Shared/LocalFileStorage.cs

```diff
@@ -0,0 +1,179 @@
+        /// Default is 50 MB.
+        /// </param>
+        /// <param name="maintenancePeriodInMilliseconds">
+        /// Maintenance event runs at specified interval.
+        /// Removes expired leases and blobs that exceed retention period.
+        /// Default is 2 minute.
+        /// </param>
+        /// <param name="retentionPeriodInMilliseconds">
+        /// Retention period in milliseconds for the blob.
+        /// Default is 2 days.
+        /// </param>
+        /// <param name="writeTimeoutInMilliseconds">
+        /// Controls the timeout when writing a buffer to blob.
+        /// Default is 1 minute.
+        /// </param>
+        public LocalFileStorage(
+                                string path,
+                                long maxSizeInBytes = 52428800,
+                                int maintenancePeriodInMilliseconds = 6000,
+                                long retentionPeriodInMilliseconds = 172800,
+                                int writeTimeoutInMilliseconds = 6000)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            this.directoryPath = PersistentStorageHelper.CreateSubdirectory(path);
+            this.maxSizeInBytes = maxSizeInBytes;
+            this.retentionPeriodInMilliseconds = retentionPeriodInMilliseconds;
+            this.writeTimeoutInMilliseconds = writeTimeoutInMilliseconds;
+
+            this.maintenanceTimer = new Timer(maintenancePeriodInMilliseconds);
+            this.maintenanceTimer.Elapsed += this.OnMaintenanceEvent;
+            this.maintenanceTimer.AutoReset = true;
+            this.maintenanceTimer.Enabled = true;
+        }
+
+        public void Dispose()
+        {
+            this.Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        public void Dispose(bool disposing)
+        {
+            if (!this.disposedValue)
+            {
+                if (disposing)
+                {
+                    this.maintenanceTimer.Dispose();
+                }
+
+                this.disposedValue = true;
+            }
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<IPersistentBlob> GetBlobs()
+        {
+            var retentionDeadline = DateTime.UtcNow - TimeSpan.FromMilliseconds(this.retentionPeriodInMilliseconds);
+
+            foreach (var file in Directory.EnumerateFiles(this.directoryPath, "*.blob", SearchOption.TopDirectoryOnly).OrderByDescending(f => f))
+            {
+                DateTime fileDateTime = PersistentStorageHelper.GetDateTimeFromBlobName(file);
+                if (fileDateTime > retentionDeadline)
+                {
+                    yield return new LocalFileBlob(file);
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public IPersistentBlob GetBlob()
+        {
+            var iterator = this.GetBlobs().GetEnumerator();
+            iterator.MoveNext();
```

Comment:
Do we need to check if `MoveNext` returned `true`?

Reply:
`MoveNext()` is not in the loop. I believe we could ignore its return type.

If `iterator` is null, both this `iterator.MoveNext()` and `iterator.Current` does not throw exception. Avoided conditional check. If iterator is null this method returns null to the caller without any checks. I will get the test case added for this scenario.